wordpress-mobile/WordPress-iOS
# Big Titles, White Headers 1

Refs #14774.

This PR is the first part of "Big Titles, White Headers". It sets up a feature flag, and updates the colours of navigation bars throughout the app. This does not yet include large titles, integrated search bars, or serif fonts.

I've set the feature flag to a8cBranchTesting so that we can do an installable build here for @osullivanchris to look at. I'll set it back to localDeveloper before merging.

![iOS 13](https://user-images.githubusercontent.com/4780/91603669-56799a00-e965-11ea-8e1b-f3442b275d10.png)

![iOS 13 Dark Mode](https://user-images.githubusercontent.com/4780/91603681-59748a80-e965-11ea-9a3a-53dd29e2c1c9.png)

![iOS 12](https://user-images.githubusercontent.com/4780/91603676-58435d80-e965-11ea-886a-8045ec809192.png)

Most of the changes here are actually to consolidate the style used for `preferredStatusBarStyle` throughout the app, so that the status bar appears the correct colour. I think we can probably even remove many of the remaining calls, as most of this should be controlled through the navigation controllers and split view controllers we use. But I think that can be cleanup for a future PR.

I also removed the `customizeAppearanceForTextElements` method from the app delegate – this was called on launch and every time the system's content size changed. I didn't see the use in calling this method multiple times, and I think that the things it configured were either unnecessary or also done elsewhere.

**To test**

- Check the code
- Build and run in:
  - iOS 12
  - iOS 13 light mode
  - iOS 13 dark mode
- Take a look through as many screens as you can in the app and check the navigation bar and its contents look correct. Look out for things like:
  - Bars created in libraries outside of the main app, like Authenticator, Media Picker, Gutenberg, and Zendesk (@ScoutHarris I tagged you here specifically for auth because I don't want to disrupt the work you've been doing there). Please check that unified and non-unified auth flows look correct.
  - Places we use search bars
  - Modal windows
  - System views like the document picker used if you add media from the 'other apps' source
- Double check things still look okay with the feature flag disabled

PR submission checklist:
